eisbuk-admin/src/components/atoms/AttendanceCard/__tests__/AttendanceCard.test.tsx

```diff
@@ -1,38 +1,43 @@
 import React from "react";
 import { render, screen } from "@testing-library/react";
 import AttendanceCard from "../AttendanceCard";
-
 import * as attendanceOperations from "@/store/actions/attendanceOperations";
-import { Category } from "eisbuk-shared";
-import { dummySlot } from "../__testData__/dummyData";
+import { Category, SlotType } from "eisbuk-shared";
+import { customersSlot } from "../__testData__/dummyData";
+import i18n from "i18next";
 
-const mockMarkAttImplementation = (customerId: string, attended: boolean) => ({
+const mockDispatch = jest.fn();
+jest.mock("react-redux", () => ({
+  useDispatch: () => mockDispatch,
+}));
+const mockMarkAttImplementation = (customerId: string, attended?: boolean) => ({
```

Comment:
Oh..the boolean here (and in the actual function should be boolean, not optional). The optional usage is "better" for components as it's often times used as a flag: `<ExpandableRegion isOpen  />`. Here the `isOpen` is optional boolean and if you put it like that, it is `true` and component will check for truthy of falsy (for rendering mostly), but in this case, we're storing the attended in a database which should always have a value. To ensure the optional `isAbsent` is one of `true` | `false` (not `null`, `undefined`, `0`, etc.) you can use `Boolean(isAbsent)` and it will cast `undefined` to `false`.